src/main/java/io/orangebeard/listener/OrangebeardExtension.java

```diff
@@ -53,6 +54,9 @@
     private final Map<String, UUID> runningTests = new HashMap<>();
     private UUID testrunUUID;
 
+    /** Tree-structure to keep track of the hierarchy of test suites. */
+    private final TestSuiteTree root = new TestSuiteTree("ROOT", null); // Only the root node should have a null UUID.
```

Comment:
I understand adding the tree. But do we still need the suite list at 52? Can't we manage with one suite structure?